### PR TITLE
Gradle upgrades

### PR DESCRIPTION
**Pull request by evan-masseau**
# Description

The initial motivation for this was to update the compile/target Android SDK to 34. I took the opportunity to add a few additional version updates to other dependencies.
Reading through semver discussions, this is arguably acceptable as a _minor_ release. Given that we are doing a major release right shortly, I think its best to include it now. 

# Check List

- [x] Are you changing anything with the public API? -- NO
- [x] Are your changes backwards compatible with previous SDK Versions? -- YES
- [x] Have you tested this change on real device? -- TODO
- [x] Have you added unit test coverage for your changes? -- YES / UPDATED
- [x] Have you verified that your changes are compatible with all Android versions the SDK currently supports? -- N/A - changing the SDK version!

## Changelog / Code Overview

Firstly, I added a gradle plugin called [refresh versions](https://splitties.github.io/refreshVersions/) to help manage the versions of our dependencies and identify when updates are available. That made a lot of updating easier. To be clear, its a gradle plugin, not a dependency of the SDK code.

Trying to update to SDK 34 triggered a chain of dependency upgrades:
- Compiling Android SDK 34 requires AGP 8.2
- AGP 8.2 requires Gradle 8.2+ (I opted to go to the latest, 8.5)
- Gradle 8+ requires JDK to update to 17
- JDK 17 requires a change to mock `SDK_INT` field in unit tests.
- `jitpack.yml` file required to run our builds on JDK 17

## Test Plan

- Pull into test app, build against a jitpack snapshot of this branch

## Related Issues/Tickets

CHNL-4510

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -23,10 +23,6 @@ android.useAndroidX=true
 # Automatically convert third-party libraries to use AndroidX
 android.enableJetifier=true
 
-# Opt-in to upcoming AGP feature to prevent building components for all variants automatically
```
Comment (evan-masseau) [inline gradle.properties:26]: Don't need to opt in anymore, this is now the default.

File: .github/workflows/android-feature.yml [hunk 2]
```diff
@@ -19,11 +19,11 @@ jobs:
       - uses: actions/checkout@v4
 
       # Sets up a JDK on the job so that we can run Java code
-      - name: Set up JDK 11
+      - name: Set up JDK 17
```
Comment (evan-masseau) [inline .github/workflows/android-feature.yml:22]: JDK 17 now required

File: build.gradle [hunk 3]
```diff
@@ -18,10 +18,10 @@ buildscript {
         // NOTE: These are dependencies for the build script itself
         // Do not place your application dependencies here;
         // they belong in the individual module build.gradle files
-        classpath "com.android.tools.build:gradle:${gradleBuildToolsVersion}"
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:${gradleKotlinVersion}"
-        classpath "org.jetbrains.dokka:dokka-gradle-plugin:${dokkaVersion}"
-        classpath "org.jlleitschuh.gradle:ktlint-gradle:${gradleKtlintVersion}"
+        classpath Android.tools.build.gradlePlugin
```
Comment (evan-masseau) [inline build.gradle:21]: This is syntax from refresh versions plugin

File: gradlew [hunk 4]
```diff
@@ -1,78 +1,127 @@
-#!/usr/bin/env sh
+#!/bin/sh
```
Comment (evan-masseau) [inline gradlew:1]: gradlew upgraded to 8.5

File: jitpack.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+jdk:
```
Comment (evan-masseau) [inline jitpack.yml:1]: config to indicate that jitpack builder must use JDK 17

File: sdk/fixtures/src/main/java/com/klaviyo/fixtures/BaseTest.kt [hunk 6]
```diff
@@ -97,21 +98,37 @@ abstract class BaseTest {
...
      * @param field
      * @param newValue
      */
     @Throws(Exception::class)
     protected fun setFinalStatic(field: Field, newValue: Any?) {
         field.isAccessible = true
-        val modifiersField: Field = Field::class.java.getDeclaredField("modifiers")
-        modifiersField.isAccessible = true
-        modifiersField.setInt(field, field.modifiers and Modifier.FINAL.inv())
+
```
Comment (evan-masseau) [inline sdk/fixtures/src/main/java/com/klaviyo/fixtures/BaseTest.kt:107]: JDK 17 fixes to mocking `SDK_INT`

File: sdk/push-fcm/build.gradle [hunk 7]
```diff
@@ -17,7 +17,8 @@ android {
 dependencies {
     implementation project(':sdk:core')
     implementation project(':sdk:analytics')
-    implementation "com.google.firebase:firebase-messaging:$ext.fcmVersion"
+    implementation platform(Firebase.bom)
```
Comment (evan-masseau) [inline sdk/push-fcm/build.gradle:20]: slight adjustment for using refreshVersions, still imports just firebase messaging.

File: versions.gradle [hunk 8]
```diff
@@ -1,33 +1,15 @@
 ext {
-    // build dependencies
-    gradleBuildToolsVersion = '7.3.1'
-    gradleKotlinVersion = '1.7.20'
+    // build dependencies - not managed by refreshVersions
```
Comment (evan-masseau) [inline versions.gradle:2]: Most dependency versions now managed in `versions.properties`

Comment (evan-masseau) [reply versions.gradle:2]: only ones it couldn't move were the build plugins for dokka and gradle ktlint... might be able to fix that later.

File: versions.properties [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+version.junit.junit=4.13.2
+
+version.androidx.test.runner=1.5.2
+##               # available=1.5.3-alpha01
+##               # available=1.6.0-alpha01
+##               # available=1.6.0-alpha02
+##               # available=1.6.0-alpha03
+##               # available=1.6.0-alpha04
+##               # available=1.6.0-alpha05
```
Comment (kennyklaviyo) [inline versions.properties:36]: theres a config setting to refreshVersions that should ignore listing alpha versions like this if you want to add that

Comment (evan-masseau) [reply versions.properties:36]: oh yes please, will do

File: settings.gradle [hunk 10]
```diff
@@ -1,3 +1,13 @@
+plugins {
+    id 'de.fayard.refreshVersions' version '0.60.3'
+}
+
 // sdk modules
 include ':sdk'
 include ':sdk:core', ':sdk:analytics', ':sdk:push-fcm', ':sdk:fixtures'
+
+refreshVersions {
```
Comment (evan-masseau) [inline settings.gradle:9]: @kennyklaviyo thanks for the tip

**Review by kennyklaviyo — APPROVED** [review]

**Pull request merged**